core/dbt/contracts/graph/model_config.py

```diff
@@ -445,6 +459,7 @@ class NodeConfig(NodeAndTestConfig):
     # sometimes getting the Union order wrong, causing serialization failures.
     unique_key: Union[str, List[str], None] = None
     on_schema_change: Optional[str] = "ignore"
+    on_configuration_change: List[OnConfigurationChangeOptions] = field(default_factory=list)
```

Comment:
Can we have more than one setting for `on_configuration_change`? I think this might just be `Optional[OnConfigurationChangeOptions]`.

Reply:
changed name, trying to use the `Optional[OnConfigurationChangeOptions]` by itself just lead to a missing type issue, trying to provide a string "apply" value lead to a type mismatch error so tried to be more inline with the other Enum related classes and used the field call to set it